### PR TITLE
libgpu/vulkan: Remove double negation

### PR DESCRIPTION
**Pull request by Narann**
Remove double negation in method arguments and checks.

### CODE REVIEW

File: src/libgpu/src/vulkan/vulkan_driver.h [hunk 1]
```diff
@@ -601,12 +601,12 @@ class Driver : public gpu::VulkanDriver, public Pm4Processor
    void _invalidateSurface(SurfaceObject *surface, SurfaceSubRange range);
    void _barrierSurface(SurfaceObject *surface, ResourceUsage usage, vk::ImageLayout layout, SurfaceSubRange range);
    SurfaceObject * getSurface(const SurfaceDesc& info);
-   void transitionSurface(SurfaceObject *surface, ResourceUsage usage, vk::ImageLayout layout, SurfaceSubRange range, bool skipChangeCheck = false);
+   void transitionSurface(SurfaceObject *surface, ResourceUsage usage, vk::ImageLayout layout, SurfaceSubRange range, bool doChangeCheck = true);
```
Comment (CodingKoopa) [inline src/libgpu/src/vulkan/vulkan_driver.h]: IMO the "do" here might be unecessary to have?

Comment (Narann) [reply src/libgpu/src/vulkan/vulkan_driver.h]: I don't know... `changeCheck` could be understands as "change the check".

Comment (CodingKoopa) [reply src/libgpu/src/vulkan/vulkan_driver.h]: What exactly is it doing then?

Comment (Narann) [reply src/libgpu/src/vulkan/vulkan_driver.h]: It checks the surface has changed.

Comment (CodingKoopa) [reply src/libgpu/src/vulkan/vulkan_driver.h]: `do` is fairly unconventional for bool names in my experience. Perhaps `checkSurfaceChanges`.

Comment (Narann) [reply src/libgpu/src/vulkan/vulkan_driver.h]: I agree, `checkSurfaceChanges` is better.

**Comment by brett19** [comment]
Original implementation follows the 'principal of least surprise'.  In the case of skipping the surface checks, the default of a boolean is typical false, and in this case, the majority case involves performing the surface checks.  This is why we added a flag for skipping as opposed to a flag for checking.

**Comment by Narann** [comment]
Thanks for the explanation Brett. I think false booleans makes code harder to read and logic flows hard to follow. Maybe it’s just me.
